src/rpc/blockchain.cpp

```diff
@@ -232,60 +258,108 @@ UniValue dumpcontracts(const UniValue& params, bool fHelp)
     int num_verified_beacons = 0;
 
     CBlock block;
-    CAutoFile file(fsbridge::fopen(path, "wb"), SER_DISK, PROTOCOL_VERSION);
-    CDataStream ss(SER_DISK, PROTOCOL_VERSION);
 
+    LOCK(cs_main);
+
+    // Set default high_height here if not specified above now that lock on cs_main is taken.
+    if (!high_height)
     {
-        LOCK(cs_main);
+        high_height = pindexBest->nHeight;
+    }
 
-        // Set default high_height here if not specified above now that lock on cs_main is taken.
-        if (!high_height)
-        {
-            high_height = pindexBest->nHeight;
-        }
+    CBlockIndex* pblockindex = pindexBest;
 
-        CBlockIndex* pblockindex = pindexBest;
+    // Rewind to high height to get time.
+    for (; pblockindex; pblockindex = pblockindex->pprev)
+    {
+        if (pblockindex->nHeight == high_height) break;
+    }
 
-        // Rewind to high height to get time.
-        for (; pblockindex; pblockindex = pblockindex->pprev)
-        {
-            if (pblockindex->nHeight == high_height) break;
-        }
+    high_height_time = pblockindex->nTime;
 
-        high_height_time = pblockindex->nTime;
+    // Continue rewinding to low height.
+    for (; pblockindex; pblockindex = pblockindex->pprev)
+    {
+        if (pblockindex->nHeight == low_height) break;
+    }
 
+    // From this point we have to go down two somewhat different paths based on whether a minimalist text output
+    // is desired or the full serialization output.
+    if (txids_only) {
```

Comment:
nit: I am not exactly sure changing the output to CSV with a flag is a good idea, I prefer returning a list of TXIDs in the JSON return. However dumpcontracts is more tailored to the developers and I don't see a point in delaying the merge and test of this PR for something trivial like this.

Reply:
I find JSON harder to work with in a bash script than a flat csv file for simple stuff. If we were unpacking complicated objects, such as the entire serializations rendered as JSON, that would be appropriate. In fact, we may very well want to do a third mode for that. We should write a helper JSON output function for that, however. That can be reserved for later.